plugins/fortinet_fortigate/icon_fortinet_fortigate/util/api.py

```diff
@@ -0,0 +1,104 @@
+import requests
+from requests import Session
+from insightconnect_plugin_runtime.exceptions import PluginException
+from icon_fortinet_fortigate.util.util import Helpers
+import json
+
+
+class FortigateAPI:
+    def __init__(self, host, api_key, logger, ssl_verify=False):
+        self.host = host
+        self.logger = logger
+        self.ssl_verify = ssl_verify
+        self.session = Session()
+        self.session.params = {"access_token": api_key}
+        self.helper = Helpers(self.logger)
+
+    def call_api(
+        self, path: str, method: str = "GET", params: dict = None, json_data: dict = None
+    ) -> requests.Response:
+        try:
+            response = self.session.request(
+                method=method.upper(),
+                url=f"https://{self.host}/api/v2/cmdb/{path}",
+                verify=self.ssl_verify,
+                json=json_data,
+                params=params,
+            )
+            self.helper.http_errors(response.text, response.status_code)
+            if response.status_code > 500:
```

Comment:
```suggestion
            if response.status_code >= 500:
```

Reply:
@blaxminarayan-r7 Status code 500 is handled by the `http_errors` method, so this change is not needed.